tools/utility.hpp

```diff
@@ -0,0 +1,146 @@
+    mapperCall.append(root);
+    mapperCall.append(depth);
+    mapperCall.append(interfaces);
+
+    MapperResponse result = {};
+
+    try
+    {
+        auto response = bus.call(mapperCall);
+
+        response.read(result);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        std::cout << "GetSubtree mapper call failed with exception: "
+                  << e.what() << std::endl;
+    }
+    return result;
+}
+
+/**
+ * @brief Templated API to set D-Bus property.
+ *
+ * @param[in] serviceName - D-Bus service name hosting the object path.
+ * @param[in] objectPath - D-Bus object path hosting the interface.
+ * @param[in] interface - D-Bus interface hosting the porperty.
+ * @param[in] property - D-Bus property to be set.
+ * @param[in] value - Value to be set.
+ */
+template <typename T>
+void setProperty(const std::string& serviceName, const std::string& objectPath,
+                 const std::string& interface, const std::string& property,
+                 const std::variant<T>& value)
+{
+    try
+    {
+        auto bus = sdbusplus::bus::new_default();
+        auto method =
+            bus.new_method_call(serviceName.c_str(), objectPath.c_str(),
+                                "org.freedesktop.DBus.Properties", "Set");
+
+        method.append(interface);
+        method.append(property);
+        method.append(value);
+        bus.call(method);
+    }
+    catch (const sdbusplus::exception::SdBusError& e)
+    {
+        std::cout << "Set property: " << property
+                  << " failed for path: " << objectPath
+                  << " with error: " << e.what() << std::endl;
+    }
+}
+
+/**
+ * @brief Templated API to get D-Bus property.
+ *
+ * @param[in] serviceName - D-Bus service name hosting the object path.
+ * @param[in] objectPath - D-Bus object path hosting the interface.
+ * @param[in] interface - D-Bus interface hosting the porperty.
+ * @param[in] property - D-Bus property whose value is to be fetched.
+ * @return Value of the property.
+ */
+template <typename T>
+T getProperty(const std::string& serviceName, const std::string& objectPath,
+              const std::string& interface, const std::string& property)
+{
+    auto bus = sdbusplus::bus::new_default();
+    auto mapperCall =
+        bus.new_method_call(serviceName.c_str(), objectPath.c_str(),
+                            "org.freedesktop.DBus.Properties", "Get");
+    mapperCall.append(interface);
+    mapperCall.append(property);
+
+    T result = {};
+
+    try
+    {
+        auto response = bus.call(mapperCall);
+
+        response.read(result);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        std::cout << "Get property: " << property
+                  << " failed for path: " << objectPath
+                  << " with error: " << e.what() << std::endl;
+    }
+
+    return result;
+}
+
+/**
+ * @brief An API to check chassis power state.
+ *
+ * @return bool - Power state of chassis, false otherwise.
```

Comment:
when true will be set is not specified.
so "Power state of chassis" is enough ?

Reply:
Added extra info.